js/src/views/Signer/components/RequestOrigin/requestOrigin.js

```diff
@@ -45,43 +46,85 @@ export default class RequestOrigin extends Component {
   renderOrigin (origin) {
     if (origin.type === 'unknown') {
       return (
-        <span className={ styles.unknown }>via unknown interface</span>
+        <span className={ styles.unknown }>
+          <FormattedMessage
+            id='signer.requestOrigin.unknownInterface'
+            defaultMessage='via unknown interface'
+          />
+        </span>
       );
     }
 
     if (origin.type === 'dapp') {
       return (
         <span>
-          by a dapp at <span className={ styles.url }>
-            { origin.details || 'unknown URL' }
-          </span>
+          <FormattedMessage
+            id='signer.requestOrigin.dapp'
+            defaultMessage='by a dapp at {url}'
+            values={ {
+              url: (
+                <span className={ styles.url }>
+                  {
+                    origin.details || (
+                      <FormattedMessage
+                        id='signer.requestOrigin.unknownUrl'
+                        defaultMessage='unknown URL'
+                      />
+                    )
+                  }
+                </span>
+              )
+            } }
+          />
         </span>
       );
     }
 
     if (origin.type === 'rpc') {
       return (
         <span>
-          via RPC <span className={ styles.url }>
-            ({ origin.details || 'unidentified' })
-          </span>
+          <FormattedMessage
+            id='signer.requestOrigin.rpc'
+            defaultMessage='via RPC {rpc}'
+            values={ {
+              url: (
+                <span className={ styles.url }>
+                  ({
+                    origin.details || (
+                      <FormattedMessage
+                        id='signer.requestOrigin.unknownRpc'
+                        defaultMessage='unidentified'
+                      />
+                    )
+                  })
+                </span>
+              )
+            } }
+          />
         </span>
       );
     }
 
     if (origin.type === 'ipc') {
       return (
         <span>
-          via IPC session
-          <span
-            className={ styles.hash }
-            title={ origin.details }
-          >
-            <IdentityIcon
-              address={ origin.details }
-              tiny
-            />
-          </span>
+          <FormattedMessage
+            id='signer.requestOrigin.ipc'
+            defaultMessage='via IPC session {icon}'
+            values={ {
+              icon: (
+                <span
+                  className={ styles.hash }
+                  title={ origin.details }
+                >
+                  <IdentityIcon
+                    address={ origin.details }
+                    tiny
+                  />
+                </span>
+              )
+            } }
+          />
```

Comment:
Yes that's true, but here it's just an `IdentityIcon`, so I guess it wouldn't make sense to put it on the right in english, and on the left in german, right?